### PR TITLE
Resolver Injection

### PR DESCRIPTION
**Pull request by fabianbs96**
Allows injecting an own `Resolver` implementation into the ctor of `LLVMBasedICFG`. ALso updates some other stuff in the resolvers

### CODE REVIEW

File: lib/PhasarLLVM/ControlFlow/LLVMBasedICFG.cpp [hunk 1]
```diff
@@ -40,11 +40,11 @@
 namespace psr {
 struct LLVMBasedICFG::Builder {
   LLVMProjectIRDB *IRDB = nullptr;
-  LLVMAliasInfoRef PT{};
+  // LLVMAliasInfoRef PT{};
```
Comment (MMory) [inline lib/PhasarLLVM/ControlFlow/LLVMBasedICFG.cpp]: just commented out?

**Review by MMory — COMMENTED** [review]
looks good to me, just one minor remark

File: lib/PhasarLLVM/ControlFlow/LLVMBasedICFG.cpp (hunk 1 above)
Comment (fabianbs96) [reply lib/PhasarLLVM/ControlFlow/LLVMBasedICFG.cpp]: fixed

**Review by MMory — APPROVED** [review]
lgtm, thanks!

**Pull request merged**